tracee-rules/benchmark/benchmark_test.go

```diff
@@ -0,0 +1,143 @@
+package benchmark
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aquasecurity/tracee/tracee-rules/benchmark/signature/golang"
+	"github.com/aquasecurity/tracee/tracee-rules/benchmark/signature/rego"
+	"github.com/aquasecurity/tracee/tracee-rules/engine"
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	inputEventsCount = 1000
+)
+
+func BenchmarkOnEventCodeInjectionRegoRule(b *testing.B) {
+	codeInjectSig, err := rego.NewCodeInjectionSignature()
+	require.NoError(b, err)
+
+	err = codeInjectSig.Init(ignoreFinding)
+	require.NoError(b, err)
+
+	for i := 0; i < b.N; i++ {
+		err = codeInjectSig.OnEvent(triggerCodeInjectorPtraceEvent)
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkOnEventCodeInjectionGoRule(b *testing.B) {
+	codeInjectSig := golang.NewCodeInjectionSignature()
+	err := codeInjectSig.Init(ignoreFinding)
+	require.NoError(b, err)
+
+	for i := 0; i < b.N; i++ {
+		err := codeInjectSig.OnEvent(triggerCodeInjectorPtraceEvent)
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkEngineWithCodeInjectionRegoRule(b *testing.B) {
+	// Prepare signatures
+	codeInjectSig, err := rego.NewCodeInjectionSignature()
+	require.NoError(b, err)
+
+	sigs := []types.Signature{
+		codeInjectSig,
+	}
+
+	for i := 0; i < b.N; i++ {
+		// Produce events
+		eventsCh := make(chan types.Event, inputEventsCount)
+		ProduceEventsInMemory(eventsCh, inputEventsCount)
+
+		inputs := engine.EventSources{
+			Tracee: eventsCh,
+		}
+		output := make(chan types.Finding, inputEventsCount)
+
+		e := engine.NewEngine(sigs, inputs, output, os.Stderr)
```

Comment:
I think we shouldn't include the engine initialization in the benchmark

Reply:
One way to do that, which I tried out in this PR is to use b.StopTimer and b.StartTime to skip timing events producer:

```go
func BenchmarkEngineWithCodeInjectionRegoRule(b *testing.B) {
	// Prepare signatures and what not

	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		b.StopTimer()
		eventsCh := make(chan types.Event, inputEventsCount)
		// Produce events without timing it
		b.StartTimer()

		// Start rules engine and wait until all events are processed
		e.Start(waitForEventsProcessed(eventsCh))

		b.Logf("Test is done with %d findings", len(output))
	}
}
```